### PR TITLE
feat: implement eth_getBlockReceipts

### PR DESCRIPTION
**Pull request by Thegaram**
## 1. Purpose or design rationale of this PR

Add `eth_getBlockReceipts`.

Original PR: https://github.com/scroll-tech/go-ethereum/pull/606

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [X] feat: A new feature

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [X] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [X] This PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Review by 0xmountaintop — DISMISSED** [review]

File: ethclient/ethclient.go [hunk 1]
```diff
@@ -101,6 +101,23 @@ func (ec *Client) BlockNumber(ctx context.Context) (uint64, error) {
 	return uint64(result), err
 }
 
+// PeerCount returns the number of p2p peers as reported by the net_peerCount method.
+func (ec *Client) PeerCount(ctx context.Context) (uint64, error) {
+	var result hexutil.Uint64
+	err := ec.c.CallContext(ctx, &result, "net_peerCount")
+	return uint64(result), err
+}
```
Comment (omerfirmak) [inline ethclient/ethclient.go:109]: is this related?

File: internal/web3ext/web3ext.go [hunk 2]
```diff
@@ -577,6 +577,22 @@ web3._extend({
...
+			name: 'getLogs',
+			call: 'eth_getLogs',
+			params: 1,
+		}),
+		new web3._extend.Method({
+			name: 'call',
+			call: 'eth_call',
+			params: 4,
+			inputFormatter: [web3._extend.formatters.inputCallFormatter, web3._extend.formatters.inputDefaultBlockNumberFormatter, null, null],
+		}),
```
Comment (omerfirmak) [inline internal/web3ext/web3ext.go:590]: how could these be missing all this time?

**Review by omerfirmak — COMMENTED** [review]
Can we get the [tests](https://github.com/ethereum/go-ethereum/commit/f1801a9feda8f81532c92077d2c9a8b785fd699b#diff-6d0f72e0732b99a2f915fe5ef1fd19f8f85a84627eceba96582ac4509c198815R2001) from the original commit as well?

File: ethclient/ethclient.go (hunk 1 above)
Comment (Thegaram) [reply ethclient/ethclient.go:109]: No. I tried and `git cherry-pick` picks up these lines for some reason. We might as well keep them.

File: internal/web3ext/web3ext.go (hunk 2 above)
Comment (Thegaram) [reply internal/web3ext/web3ext.go:590]: Yeah, I'm also surprised. This is only for the interactive geth console, I don't recall ever using these two APIs on the console. Apparently these were added to upstream after we forked off.

**Comment by Thegaram** [comment]
> Can we get the [tests](https://github.com/ethereum/go-ethereum/commit/f1801a9feda8f81532c92077d2c9a8b785fd699b#diff-6d0f72e0732b99a2f915fe5ef1fd19f8f85a84627eceba96582ac4509c198815R2001) from the original commit as well?

You're right, we should add these. But currently none of the API tests exist in our fork, and many interfaces have changed, so it's not trivial to add these. I suggest we add them in another PR to keep things clean.

File: internal/web3ext/web3ext.go (hunk 2 above)
Comment (omerfirmak) [reply internal/web3ext/web3ext.go:590]: Ah, I thought this affected RPC as well.

**Review by omerfirmak — DISMISSED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Review by omerfirmak — APPROVED** [review]
